Automattic/sensei
# Add skeleton for wizard steps.

Fixes #5126

### Changes proposed in this Pull Request
* Created `Wizard` component that expects a `steps` attribute containing an ordered array of the different `Steps` that should be rendered.
* Some sample `Steps` have been added in order to test the different cases and flows we expect to support:
  * `course-details-step.js`: Has a sample input and way to persist data in the `Wizard`'s data. It has also a sample action that displays the contents of the `Wizard` data.
  * `upgrade-step.js`: Empty step with only two actions: next and go to some url.
  * `course-patterns-step.js`: Empty step with only a final action - `onCompletion` can be called when choosing the pattern at some point.
  * `lesson-details-step.js` and `lesson-patterns-step.js` are also empty steps that are not currently used. 

### Testing instructions
- Create a new Lesson or Course and you will be prompted with a sample **wizard for Course**.
- Play around with the actions.
- Check the code :)


### Screenshot / Video
https://user-images.githubusercontent.com/799065/169290375-2e84138d-aa37-4a16-a457-4dd1fb33f025.mp4

### Discussion / Context
- This is only a proposal for the basic structure for the Wizard.
- Tried by using `{children}` but couldn't figure out the best way to have different "actions" per step - if anyone wants to give it a try I will be pleased to learn how to do that!
- I expect this task to include some minor CSS for the modal, but specific details for each step will be handled in each steps task.
- I will look how to have different steps for different post types but I want to get fast feedback on the main structure. Will update the PR with that once I have it.

### TODO
- [x] Allow different steps depending on post type.
- [x] CSS for modal + wizard structure.
- [x] Update modal title? Maybe title can be part of each step.
- [x] Add tests for `wizard.js`.
